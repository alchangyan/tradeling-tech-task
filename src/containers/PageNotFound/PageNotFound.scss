.page-not-found {
  text-align: center;
  color: #5c5a5a;

  &__background {
    display: inline-block;

    & .emoji {
      width: 300px;
      height: 300px;
      border: 20px solid #5c5a5a;
      border-radius: 50%;
      display: flex;
      flex-direction: column;
      align-items: center;

      &__eyes {
        margin-top: 70px;
        height: 50px;
        display: inline-flex;
      }

      &__eye {
        display: inline-block;
        width: 50px;
        height: 50px;
        background-color: #5c5a5a;
        border-radius: 50%;

        &:first-of-type {
          margin-right: 60px;
        }
      }

      &__mouth {
        width: 130px;
        height: 130px;
        border: 20px solid transparent;
        border-right: 20px solid #5c5a5a;
        border-top: 20px solid #5c5a5a;
        border-radius: 50%;
        transform: rotate(-45deg) translate(-22px, 22px);
        position: relative;

        &::after,
        &::before {
          content: '';
          position: absolute;
          border-radius: 50%;
          width: 20px;
          height: 20px;
          background-color: #5c5a5a;
        }

        &::after {
          top: 2px;
          left: 2px;
        }

        &::before {
          left: 108px;
          top: 108px;
        }
      }
    }
  }

  &__404 {
    font-size: 140px;
    font-weight: 900;
  }
}
