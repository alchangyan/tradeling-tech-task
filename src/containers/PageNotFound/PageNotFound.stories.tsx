import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import PageNotFound from './PageNotFound';

export default {
  title: 'Containers/PageNotFound',
  component: PageNotFound,
  parameters: {
    jest: ['PageNotFound'],
  },
} as Meta;

const Template: Story<{}> = args => <PageNotFound />;

export const Default = Template.bind({});
Default.args = {};
