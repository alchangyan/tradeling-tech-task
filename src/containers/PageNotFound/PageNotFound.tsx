import * as React from 'react';
import { useHistory } from 'react-router-dom';
import Header from '../../components/Header';

import './PageNotFound.scss';

const PageNotFound: React.FC<{}> = () => {
  const history = useHistory();
  const handleRedirect = React.useCallback(() => history && history.push('/'), [
    history,
  ]);

  return (
    <div className="page-not-found">
      <Header />
      <div className="page-not-found__background">
        <div className="emoji">
          <div className="emoji__eyes">
            <div className="emoji__eye"></div>
            <div className="emoji__eye"></div>
          </div>
          <div className="emoji__mouth" />
        </div>
      </div>
      <div className="page-not-found__404">404</div>
      <h2 className="page-not-found__title">Page Not Found...</h2>
      <h4 className="page-not-found__subtitle">
        ...so, you still have a chance to make one!!!
      </h4>
      <button onClick={handleRedirect}>or just go to homepage</button>
    </div>
  );
};

export default PageNotFound;
