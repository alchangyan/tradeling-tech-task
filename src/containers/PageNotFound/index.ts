import PageNotFound, { TPageNotFoundProps } from "./PageNotFound";

export type { TPageNotFoundProps };
export default PageNotFound;
