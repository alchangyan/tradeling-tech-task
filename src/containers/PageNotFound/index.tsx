import * as React from 'react';
import PageNotFound from './PageNotFound';

export default React.memo(PageNotFound);
