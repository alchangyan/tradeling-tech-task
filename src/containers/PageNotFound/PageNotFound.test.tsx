import React from 'react';
import {
  render,
  cleanup,
  RenderResult,
  fireEvent,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import PageNotFound from './index';
const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

interface TGetComponentResult {
  wrapper: RenderResult;
  container: HTMLElement;
  comp: HTMLElement;
}

let comp: HTMLElement;
let wrapper: RenderResult;
let container: HTMLElement;

const getComponent = (): TGetComponentResult => {
  wrapper = render(
    <MemoryRouter>
      <PageNotFound />
    </MemoryRouter>,
  );

  ({ container } = wrapper);
  comp = container.firstChild as HTMLElement;

  return {
    wrapper,
    container,
    comp,
  };
};

beforeEach(cleanup);
afterEach(cleanup);

describe('PageNotFound', () => {
  it('renders correctly', () => {
    getComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls history.push', () => {
    getComponent();
    fireEvent.click(wrapper.getByRole('button'));
    expect(mockHistoryPush).toHaveBeenCalledWith('/');
  });
});
