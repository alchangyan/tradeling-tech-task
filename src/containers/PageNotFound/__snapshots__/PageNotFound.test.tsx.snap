// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound renders correctly 1`] = `
<div
  class="page-not-found"
>
  <div
    class="header"
  >
    <img
      alt="github logo"
      class="header__logo"
      src="/octacat.png"
    />
    <div
      class="header__content"
    >
      <h3
        class="header__title"
      >
        GitHub Searcher
      </h3>
      <div
        class="header__subtitle"
      >
        Search users or repositories below
      </div>
    </div>
  </div>
  <div
    class="page-not-found__background"
  >
    <div
      class="emoji"
    >
      <div
        class="emoji__eyes"
      >
        <div
          class="emoji__eye"
        />
        <div
          class="emoji__eye"
        />
      </div>
      <div
        class="emoji__mouth"
      />
    </div>
  </div>
  <div
    class="page-not-found__404"
  >
    404
  </div>
  <h2
    class="page-not-found__title"
  >
    Page Not Found...
  </h2>
  <h4
    class="page-not-found__subtitle"
  >
    ...so, you still have a chance to make one!!!
  </h4>
  <button>
    or just go to homepage
  </button>
</div>
`;
